plugins/crd/plugin_impl_crd_test.go

```diff
@@ -0,0 +1,208 @@
+	flag.Parse()
+	kubeconfig = *kc
+}
+
+func k8sClient(kc string) (*kubernetes.Clientset, *apiextcs.Clientset, *networkservicemesh.Clientset, error) {
+	cfg, err := clientcmd.BuildConfigFromFlags("", kc)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	kubeClient, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	apiextClient, err := apiextcs.NewForConfig(cfg)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	crdClient, err := networkservicemesh.NewForConfig(cfg)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	return kubeClient, apiextClient, crdClient, nil
+}
+
+func setupEnv(k8s *kubernetes.Clientset, apiextClient *apiextcs.Clientset) error {
+	crds := []struct {
+		name     string
+		plural   string
+		fullname string
+	}{
+		{name: reflect.TypeOf(v1.NetworkServiceEndpoint{}).Name(), plural: v1.NSMEPPlural, fullname: v1.FullNSMEPName},
+		{name: reflect.TypeOf(v1.NetworkServiceChannel{}).Name(), plural: v1.NSMChannelPlural, fullname: v1.FullNSMChannelName},
+		{name: reflect.TypeOf(v1.NetworkService{}).Name(), plural: v1.NSMPlural, fullname: v1.FullNSMName},
+	}
+	plugin := Plugin{
+		k8sClientset: k8s,
+		apiclientset: apiextClient,
+	}
+	// Setting up testing namespace
+	namespace := corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: nsmTestNamespace,
+		},
+	}
+	if _, err := k8s.CoreV1().Namespaces().Create(&namespace); err != nil {
+		return err
+	}
+	// Creating CRD definitions
+	for _, crd := range crds {
+		err := createCRD(&plugin, crd.fullname,
+			v1.NSMGroup,
+			v1.NSMGroupVersion,
+			crd.plural,
+			crd.name)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func cleanupEnv(k8s *kubernetes.Clientset, apiextClient *apiextcs.Clientset) error {
+
+	crds := []struct {
+		name     string
+		plural   string
+		fullname string
+	}{
+		{name: reflect.TypeOf(v1.NetworkServiceEndpoint{}).Name(), plural: v1.NSMEPPlural, fullname: v1.FullNSMEPName},
+		{name: reflect.TypeOf(v1.NetworkServiceChannel{}).Name(), plural: v1.NSMChannelPlural, fullname: v1.FullNSMChannelName},
+		{name: reflect.TypeOf(v1.NetworkService{}).Name(), plural: v1.NSMPlural, fullname: v1.FullNSMName},
+	}
+	for _, crd := range crds {
+		err := apiextClient.ApiextensionsV1beta1().CustomResourceDefinitions().Delete(crd.fullname, &metav1.DeleteOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	if err := k8s.CoreV1().Namespaces().Delete(nsmTestNamespace, &metav1.DeleteOptions{}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func TestCRDValidation(t *testing.T) {
+
+	if kubeconfig == "" {
+		t.Skip("This test requires a valid kubeconfig file, skipping...")
+	}
+	k8sClient, apiextClient, crdClient, err := k8sClient(kubeconfig)
+	if err != nil {
+		t.Errorf("Fail to get k8s client with error: %+v", err)
+	}
+	if err := setupEnv(k8sClient, apiextClient); err != nil {
+		t.Errorf("Fail to setup test environment with error: %+v", err)
+	}
+	defer func() {
```

Comment:
Love the usage of defer() here, it's one of the things I like about go. :)

Reply:
Yeah, I really like it too!!